tests/test_smoke_tests.py

```diff
@@ -66,16 +92,110 @@ def is_compatible(pyenv_version: str) -> bool:
                 f"Python {python_version} is not installed via Pyenv. Please install with "
                 f"`pyenv install`. All installed versions: {', '.join(self.pyenv_versions)}."
             )
+
+        if alias:
+            with self._setup_alias(compatible_pyenv_version, alias, bad_aliases) as python_setup:
+                yield python_setup
+        else:
+            with self._run_pyenv_local(compatible_pyenv_version):
+                yield self.PythonSetup()
+
+    @staticmethod
+    def which(*exe_names: str, fallible: bool = False) -> Mapping[str, Path]:
+        return {
+            exe.name: exe
+            for exe in map(
+                Path,
+                subprocess.run(
+                    [
+                        "/usr/bin/env",
+                        "bash",
+                        "-c",
+                        f"command -v {' '.join(shlex.quote(exe_name) for exe_name in exe_names)}",
+                    ],
+                    check=not fallible,
+                    stdout=subprocess.PIPE,
+                )
+                .stdout.decode()
+                .splitlines(),
```

Comment:
Would `shutil.which()` work? https://docs.python.org/3/library/shutil.html#shutil.which

Reply:
Calling it 20+ times would, but not once like this to get everything.